### ISSUE TITLE
placesCenter@scollins - Invalid search.py permissions

### ISSUE THREAD

**Issue by pjobson** [linuxmint/cinnamon-spices-applets#1859]
```
 * App Version - 2.1 / 2018-03-26 22:30 UTC
 * Cinnamon Version - 3.6.7
 * Distribution - Linux Mint 18.3 Sylvia
 ```

**Notify author of applet**
@collinss 

**Issue**

Invalid permissions in search.py

**Steps to reproduce**

Click the folder icon then magnifying glass.

[Screen Cap](https://i.imgur.com/GQbFduc.png)

**Solution**

    chmod +x ~/.local/share/cinnamon/applets/placesCenter@scollins/search.py

**Comment by collinss** [comment]
Hi @pjobson, this is caused by a regression in Cinnamon 3.6. The problem is that the library that cinnamon-settings uses to extract the zip file from the spices site doesn't preserve file permissions, and the previous workaround wasn't included when the code in question was rewritten. It has been fixed in Cinnamon 3.8, but unfortunately there is nothing that can be done for 3.6 besides manually editing the permissions as you did.

**Comment by claudiux** [comment]
If you want, you can add, at the beginning of the _init function of MyApplet, the line:
`GLib.spawn_command_line_async("bash -c 'cd "+ metadata.path + " && chmod 755 search.py'");`

Thus, every time this applet starts, the right permissions are set to the search.py script.
